File: frontend/src/services/comparisonService.ts
import api from './api';
import { Comparison, CreateComparisonRequest, ApiResponse } from '@/types';

export const comparisonService = {
  createComparison: async (request: CreateComparisonRequest): Promise<Comparison> => {
    const response = await api.post<ApiResponse<Comparison>>('/comparisons', request);
    return response.data.data;
  },

  getComparisonsByCalibrationId: async (calibrationId: number): Promise<Comparison[]> => {
    const response = await api.get<ApiResponse<Comparison[]>>(`/comparisons/calibration/${calibrationId}`);
    return response.data.data;
  },

  getPendingComparisons: async (calibrationId: number): Promise<Comparison[]> => {
    const response = await api.get<ApiResponse<Comparison[]>>(`/comparisons/calibration/${calibrationId}/pending`);
    return response.data.data;
  },
};

File: backend/src/main/java/com/grading/service/ComparisonService.java
package com.grading.service;

import com.grading.dto.request.CreateComparisonRequest;
import com.grading.dto.response.ComparisonResponse;

import java.util.List;

public interface ComparisonService {
    ComparisonResponse createComparison(CreateComparisonRequest request, Long decidedById);
    List<ComparisonResponse> getComparisonsByCalibrationId(Long calibrationId);
    List<ComparisonResponse> getPendingComparisonsForEvaluator(Long calibrationId, Long evaluatorId);
}



